File: src/cn/yiiguxing/plugin/translate/action/ExclusiveTranslateAction.java
package cn.yiiguxing.plugin.translate.action;

import org.jetbrains.annotations.NotNull;

/**
 * 自动取单个词，忽略选择
 */
public class ExclusiveTranslateAction extends TranslateAction {

    public ExclusiveTranslateAction() {
        super(false);
    }

    @NotNull
    @Override
    protected AutoSelectionMode getAutoSelectionMode() {
        return AutoSelectionMode.EXCLUSIVE;
    }
}

File: src/main/kotlin/cn/yiiguxing/plugin/translate/action/ExclusiveTranslateAction.kt
package cn.yiiguxing.plugin.translate.action

import cn.yiiguxing.plugin.translate.adaptedMessage
import cn.yiiguxing.plugin.translate.message
import cn.yiiguxing.plugin.translate.util.SelectionMode

/**
 * 翻译动作，自动取单个词，忽略选择
 */
class ExclusiveTranslateAction : TranslateAction(false) {

    init {
        isEnabledInModalContext = true
        templatePresentation.text = adaptedMessage("action.ExclusiveTranslateAction.text")
        templatePresentation.description = message("action.description.exclusive")
    }

    override val selectionMode: SelectionMode
        get() = SelectionMode.EXCLUSIVE

}
